app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -3210,25 +3267,58 @@ class BrowserTabViewModel @Inject constructor(
     }
 
     override fun onReceivedMaliciousSiteWarning(url: Uri, feed: Feed, exempted: Boolean, clientSideHit: Boolean) {
-        site = siteFactory.buildSite(url = url.toString(), tabId = tabId)
-        site?.maliciousSiteStatus = when (feed) {
+        val maliciousSiteStatus = when (feed) {
             MALWARE -> MaliciousSiteStatus.MALWARE
             PHISHING -> MaliciousSiteStatus.PHISHING
         }
+        buildSiteFactory(url = url.toString(), maliciousSiteStatus = maliciousSiteStatus)
+
         if (!exempted) {
             val params = mapOf(CATEGORY_KEY to feed.name.lowercase(), CLIENT_SIDE_HIT_KEY to clientSideHit.toString())
             pixel.fire(AppPixelName.MALICIOUS_SITE_PROTECTION_ERROR_SHOWN, params)
             loadingViewState.postValue(
-                currentLoadingViewState().copy(isLoading = false, progress = 100, url = url.toString()),
+                currentLoadingViewState().copy(
+                    isLoading = false,
+                    progress = 100,
+                    url = url.toString(),
+                    privacyOn = false,
+                ),
             )
             browserViewState.postValue(
                 currentBrowserViewState().copy(
                     browserShowing = false,
+                    showPrivacyShield = HighlightableButton.Gone,
```

Comment:
You have `BrowserStateModifier.copyForHomeShowing`, you can rely on that for most of the state changes, add yours there, etc.